lizrd/core/llm.py

```diff
@@ -374,15 +357,43 @@ def get_current_device(self, block_num):
 
 
 @ash.check("... -> ... d")
-def TokenEmbedding(vocab_size, embedding_dim):
+def TokenEmbedding(
+    vocab_size,
+    embedding_dim,
+    init_type: Literal["uniform", "truncated_normal"] = "truncated_normal",
+    init_scale: float = 1.0,
+):
+    embedding = nn.Embedding(vocab_size, embedding_dim)
+    default_weight = embedding.weight.data
+    embedding.weight.data = get_init_weight(
+        shape=default_weight.shape,
+        fan_in=1,
```

Comment:
maybe leave some kind of comments that this is how its done in pytorch nn.Embedding